src/UI/Art/SpriteArtist/PlayerSetupArtist.java

```diff
@@ -0,0 +1,282 @@
+      {
+        PlayerSetupColorFactionArtist.draw(g, subMenu);
+      }
+      if( myControl.getHighlightedCategory() == PlayerSetupController.SelectionCategories.TEAM.ordinal() )
+      {
+        PlayerSetupTeamArtist.draw(g, mapInfo, subMenu);
+      }
+      if( myControl.getHighlightedCategory() == PlayerSetupController.SelectionCategories.AI.ordinal() )
+      {
+        PlayerSetupAiArtist.draw(g, subMenu, control.getPlayerInfo(control.getHighlightedPlayer()).getCurrentColor());
+      }
+    }
+  }
+
+  private static void drawPlayerSetup(Graphics g, MapInfo mapInfo, PlayerSetupController control)
+  {
+    // Get the draw space. We'll draw it all in real size and then scale it when we draw to the window.
+    int drawScale = SpriteOptions.getDrawScale();
+    Dimension dimensions = SpriteOptions.getScreenDimensions();
+    int imageWidth = dimensions.width / drawScale;
+    int imageHeight = dimensions.height / drawScale;
+    BufferedImage image = SpriteLibrary.createTransparentSprite(imageWidth, imageHeight);
+    Graphics myG = image.getGraphics();
+
+    /////////////////// Ready Button ////////////////////////
+    BufferedImage readyButton = SpriteUIUtils.makeTextFrame("Ready!", 3, 2);
+    int readyX = imageWidth-(int)(readyButton.getWidth()*1.25);
+    int readyY = imageHeight/2-readyButton.getHeight()/2;
+    myG.drawImage(readyButton, readyX, readyY, null);
+    int readyAreaWidth = (int)(readyButton.getWidth() * 1.5);
+
+    /////////////////// Player Panels ///////////////////////
+    int numCOs = mapInfo.getNumCos();
+    int highlightedPlayer = myControl.getHighlightedPlayer();
+
+    // Calculate the vertical space each player panel will consume.
+    int panelHeight = PlayerPanel.PANEL_HEIGHT+3;
+
+    // If we are moving from one option to another, calculate the intermediate draw location.
+    if( animHighlightedPlayer != highlightedPlayer )
+    {
+      double slide = SpriteUIUtils.calculateSlideAmount(animHighlightedPlayer, highlightedPlayer);
+      animHighlightedPlayer += slide;
+    }
+
+    // Define the space to draw the list of player CO portraits.
+    int playerXCenter = ((imageWidth - readyAreaWidth) / 2);
+    int highlightedPlayerYCenter = imageHeight / 2; // Whichever player has focus should be centered.
+
+    // Find where the zeroth player CO should be drawn.
+    // Shift from the center location by the spacing times the number of the highlighted option.
+    int playerYCenter = (int)(highlightedPlayerYCenter - (animHighlightedPlayer * panelHeight));
+
+    // Draw all of the player info.
+    for(int i = 0; i < numCOs; ++i, playerYCenter += (panelHeight))
+    {
+      // Only bother to draw it if it is on-screen.
+      if( (playerYCenter > -panelHeight/2) && ( playerYCenter < SpriteOptions.getScreenDimensions().getHeight()+(panelHeight/2) ) )
+      {
+        PlayerSetupInfo playerInfo = myControl.getPlayerInfo(i);
+        Integer key = new Integer(i);
```

Comment:
Deprecated constructor, you can just use `= i;`

Reply:
Well, yeah. I think I did that in other places. My hands just kinda do what they want some times.